### PR TITLE
Apply deterministic virtual_op numbering rule

### PR DESCRIPTION
**Pull request by abitmore**
https://github.com/bitshares/bitshares-core/issues/1675

### CODE REVIEW

File: libraries/chain/include/graphene/chain/operation_history_object.hpp [hunk 1]
```diff
@@ -61,7 +61,7 @@ namespace graphene { namespace chain {
          /** the operation within the transaction */
          uint16_t          op_in_trx = 0;
          /** any virtual operations implied by operation in block */
-         uint16_t          virtual_op = 0;
+         uint32_t          virtual_op = 0;
```
Comment (abitmore) [inline libraries/chain/include/graphene/chain/operation_history_object.hpp]: Reason for this change: it's not too hard to create more than 65536 virtual operations in a block.

**Comment by pmconrad** [comment]
Hm, I think you're resetting to 0 in too many places...
It should be sufficient to reset `_current_virtual_op` in `apply_transaction` before each `apply_operation`, and reset `_current_op` and `_current_virtual_op` in `apply_block` after applying all transactions.

**Comment by abitmore** [comment]
Force-pushed.

**Comment by abitmore** [comment]
Merged develop branch to resolve conflict on DB_VERSION. Please review again.

**Pull request merged**